ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala

```diff
@@ -0,0 +1,264 @@
+import com.daml.ledger.participant.state.v2.{
+  PruningResult,
+  ReadService,
+  SubmissionResult,
+  SubmitterInfo,
+  TransactionMeta,
+  Update,
+  WriteService,
+}
+import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.engine.Engine
+import com.daml.lf.transaction.SubmittedTransaction
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.akkastreams.dispatcher.{Dispatcher, SubSource}
+import com.daml.ports.Port
+import com.daml.telemetry.TelemetryContext
+import com.google.rpc.status.{Status => StatusProto}
+import io.grpc.Status.Code
+import io.grpc.{Channel, ManagedChannelBuilder, Status}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+import scala.collection.mutable
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.existentials
+
+class RunnerSpec extends AsyncWordSpec with Matchers {
+  private implicit val resourceContext: ResourceContext = ResourceContext(ExecutionContext.global)
+  private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
+
+  "the runner" should {
+    "start a participant" in {
+      val config = Config.createDefault(()).copy(ledgerId = LedgerId)
+      val participantConfig = newTestPartcipantConfig()
+      val runner = new Runner(Name, TestLedgerFactory, ConfigProvider.ForUnit)
+
+      newApp(config, participantConfig, runner).use { channel =>
+        LedgerIdentityServiceGrpc
+          .stub(channel)
+          .getLedgerIdentity(GetLedgerIdentityRequest.of())
+          .map { response =>
+            response.ledgerId should be(LedgerId)
+          }
+      }
+    }
+  }
+}
+
+object RunnerSpec {
+  private val Name = classOf[RunnerSpec].getSimpleName
+  private val LedgerId = s"$Name-Ledger"
+
+  private val logger = ContextualizedLogger.get(getClass)
+
+  private val engine = Engine.StableEngine()
+
+  private def newTestPartcipantConfig() = {
+    val participantId = Ref.ParticipantId.assertFromString("participant")
+    ParticipantConfig(
+      mode = ParticipantRunMode.Combined,
+      participantId = participantId,
+      shardName = None,
+      address = None,
+      port = Port.Dynamic,
+      portFile = None,
+      serverJdbcUrl = ParticipantConfig.defaultIndexJdbcUrl(participantId),
+      indexerConfig = ParticipantIndexerConfig(allowExistingSchema = false),
+    )
+  }
+
+  private def newApp[T <: ReadWriteService](
+      config: Config[Unit],
+      participantConfig: ParticipantConfig,
+      runner: Runner[T, Unit],
+  )(implicit
+      loggingContext: LoggingContext
+  ): ResourceOwner[Channel] =
+    for {
+      actorSystem <- ResourceOwner.forActorSystem(() => ActorSystem(Name))
+      materializer <- ResourceOwner.forMaterializer(() => Materializer(actorSystem))
```

Comment:
Don't we have an `Akka` support spec that provides these resources?

Reply:
Good call, that does make it more pleasant.